plone/app/z3cform/converters.py

```diff
@@ -303,3 +305,23 @@ def toFieldValue(self, value):
         if not value:
             return self.field.missing_value
         return value
+
+
+@adapter(IBool, ISingleCheckBoxBoolWidget)
+class BoolSingleCheckboxDataConverter(BaseDataConverter):
+    """Special converter between boolean fields and single checkbox widgets.
+    """
+
+    def toWidgetValue(self, value):
+        """Convert from Python bool to token sequence representation."""
+        if value:
+            return ['selected']
+        return ['unselected']
+
+    def toFieldValue(self, value):
+        """See interfaces.IDataConverter"""
+        # consider all different from true as false,
+        # this way it works with one checkbox
+        if value and value[0] == 'selected':
+            return True
+        return False
```

Comment:
Whether we set up the display value here or in the widget, we should probably use messageids rather than hardcoded strings.